packages/magic/src/login-modal.ts

```diff
@@ -0,0 +1,98 @@
+import { firstValueFrom, Subject, take } from 'rxjs'
+import LoginModal from './view/LoginModal.svelte'
+import { loggedIn$ } from './streams'
+import { SofiaProRegular, SofiaProSemiBold, SofiaProLight } from '@web3-onboard/common'
+import type { LoginOptions } from './types'
+
+// eslint-disable-next-line max-len
+const loginModal = async (
+  options: LoginOptions
+): Promise<boolean> => {
+  if (options) {
+    const error = !options
+
+    if (error) {
+      throw error
+    }
+  }
+
+  const app = mountLoginModal(options, loggedIn$)
+
+  loggedIn$.pipe(take(1)).subscribe(() => {
+    app.$destroy()
+  })
+
+  return firstValueFrom(loggedIn$)
+}
+
+// eslint-disable-next-line max-len
+const mountLoginModal = (
+  loginOptions: LoginOptions,
+  loggedIn$: Subject<boolean>
+) => {
+  class loginModalEl extends HTMLElement {
+    constructor() {
+      super()
+    }
+  }
+
+  if (!customElements.get('login-modal')) {
+    customElements.define('login-modal', loginModalEl)
+  }
```

Comment:
```suggestion
  if (!customElements.get('onboard-magic-login-modal')) {
    customElements.define('onboard-magic-login-modal', loginModalEl)
  }
```
To be safe I reckon we need to have a more specific namespace for this element.